src/util/fs.js

```diff
@@ -162,6 +172,56 @@ export const fileDatesEqual = (a: Date, b: Date) => {
   return aTime === bTime;
 };
 
+// This ensured the timestamps are preserved from the file in the cache to the file copied to node_modules.
+// These timestamps are checked to see if files have changed and need recopied, so preserving them is important.
+// There are some weird cases here:
+// * On linux, fs.copyFile does not preserve timestamps, but does on OSX and Win.
+// * On windows, you must open a file with write permissions to call `fs.futimes`.
+//   On OSX you can open with read permissions and still call `fs.futimes`.
+//   We first try to open the file in write mode because that works across all OSs.
+//   However if the file is read-only (not even the owner has write perms) then that will fail.
+//   We can try to reopen in read mode, which will work on OSX.
```

Comment:
It seems a bit convoluted. Would it make sense to use a third-party package for this?

Reply:
is there a 3rd party package that will handle this? I agree that it's ugly. Maybe I can move it to it's own package.

Reply:
Let's keep it there for now 👍